chain-signatures/node/src/rpc.rs

```diff
@@ -297,6 +311,44 @@ impl ContractStateWatcher {
         }
     }
 
+    /// Waits for the contract to be in Running state and extracts governance info.
+    /// Keeps waiting if we are not a participant in the current running state.
+    pub async fn wait_governance(&mut self) -> GovernanceInfo {
+        loop {
+            let running = self.wait_running().await;
+            if let Some(me) = running.participants.find_participant(&self.account_id) {
+                return GovernanceInfo {
+                    me: *me,
+                    threshold: running.threshold,
+                    epoch: running.epoch,
+                    public_key: PublicKey::from(running.public_key),
+                    participants: running.participants.keys().copied().collect(),
+                };
+            }
```

Comment:
nit: I think something like this would work, too

```suggestion
            if let Some(gov) = self.governance() {
                return gov;
            }
```